safe/gui/tools/minimum_needs/needs_calculator_dialog.py

```diff
@@ -61,148 +102,62 @@ def __init__(self, parent=None):
-            index, QtCore.Qt.UserRole)
-        # noinspection PyArgumentList
-        layer = QgsMapLayerRegistry.instance().mapLayer(layer_id)
-        fields = layer.pendingFields()
-        self.cboFields.clear()
-        has_fields = False
-        for field in fields:
-            # LOGGER.info(field.typeName())
-            # TODO exclude dates too? TS
-            if field.typeName() != 'String':
-                has_fields = True
-                add_ordered_combo_item(
-                    self.cboFields, field.name(), field.name())
-
-        # Now disable the run button if no suitable fields were found
-        # see #2206
-        ok_button = self.button_box.button(QtGui.QDialogButtonBox.Ok)
-        if not has_fields:
-            ok_button.setEnabled(False)
-        else:
-            ok_button.setEnabled(True)
+        # count each minimum needs for every features
+        for needs in minimum_needs_post_processors:
+            print needs
+            run_single_post_processor(input_layer, needs)
 
     def accept(self):
         """Process the layer and field and generate a new layer.
 
         .. note:: This is called on OK click.
 
         """
-        index = self.cboFields.currentIndex()
-        field_name = self.cboFields.itemData(
-            index, QtCore.Qt.UserRole)
-
-        index = self.cboPolygonLayers.currentIndex()
-        layer_id = self.cboPolygonLayers.itemData(
-            index, QtCore.Qt.UserRole)
-        # noinspection PyArgumentList
-        layer = QgsMapLayerRegistry.instance().mapLayer(layer_id)
-
-        file_name = layer.source()
+        # create memory layer
+        output_layer = "%s_minimum_needs" \
+                       % self.map_layer_cb.currentLayer().name()
+        output_layer = create_memory_layer(
+            output_layer,
+            self.map_layer_cb.currentLayer().geometryType(),
+            self.map_layer_cb.currentLayer().crs(),
+            self.map_layer_cb.currentLayer().fields())
+
+        # read keywords from input
+        temp_layer = self.map_layer_cb.currentLayer()
+        try:
+            keyword_io = KeywordIO()
+            temp_layer.keywords = keyword_io.read_keywords(temp_layer)
+        except KeywordNotFoundError:
+            temp_layer.keywords['layer_purpose'] = 'aggregation'
+
+        # add keywords to output layer
+        copy_layer(temp_layer, output_layer)
+        monkey_patch_keywords(output_layer)
+        output_layer.keywords["layer_purpose"] = "aggregation"
+        output_layer.keywords["inasafe_fields"] = \
+            {"displaced_field": str(self.displaced_cb.currentField()),
+             "aggregation_name_field": str(self.aggregation_cb.currentField())}
+
+        # remove unnecessary fields & rename inasafe fields
+        _rename_remove_inasafe_fields(output_layer)
 
         try:
-            output_layer = self.minimum_needs(input_layer, field_name)
+            self.minimum_needs(output_layer)
         except ValueError:
```

Comment:
I think you can remove this ValueError.
Maybe it's better to catch `Exception` in case if there is an error and to display an error message.